lms/djangoapps/program_enrollments/models.py

```diff
@@ -177,8 +177,11 @@ def change_status(self, status):
         if self.course_enrollment:
             if status == CourseEnrollmentResponseStatuses.ACTIVE:
                 self.course_enrollment.activate()
-            else:
+            elif status == CourseEnrollmentResponseStatuses.INACTIVE:
                 self.course_enrollment.deactivate()
+            else:
+                message = "Changed {enrollment} status to {status}, not changing status of its CourseEnrollment"
```

Comment:
#NIT I would like to see an explanation in the log here on why the status is not changed.

Reply:
added that, and tests for these two logging messages